### PR TITLE
End to end testing: create documentation

### PR DESCRIPTION
**Pull request by JulienBalestra**
### What does this PR do?

Create a documentation for the end to end testing.

### CODE REVIEW

File: test/e2e/README.md [hunk 1]
```diff
@@ -1,11 +1,43 @@
...
+${GOPATH}/src/github.com/DataDog/datadog-agent $ aws-vault exec ${DEV} -- inv -e e2e-tests -t dev --image datadog/agent-dev:master
+$ echo $?
+```
+
+### Locally
+
+```bash
+$ sudo pupernetes daemon run /opt/sandbox --job-type systemd 
+${GOPATH}/src/github.com/DataDog/datadog-agent $ inv -e e2e-tests -t local --image datadog/agent-dev:master
+$ echo $?
```
Comment (devonboyer) [inline test/e2e/README.md]: Did you mean for this to be

```
$ cd {GOPATH}/src/github.com/DataDog/datadog-agent && inv -e e2e-tests -t local --image datadog/agent-dev:master
```

You could also add

```
cd ${GOPATH}/src/github.com/DataDog && git clone https://github.com/DataDog/datadog-agent.git
```

File: test/e2e/README.md [hunk 2]
```diff
@@ -1,11 +1,124 @@
...
+```bash
+$ sudo pupernetes daemon run /opt/sandbox --job-type systemd 
+$ cd ${GOPATH}/src/github.com/DataDog/datadog-agent 
+$ inv -e e2e-tests -t local --image datadog/agent-dev:master
+$ echo $?
+```
+
+# Argo workflow
+
+The argo documentation is available [here](https://applatix.com/open-source/argo/docs/examples.html), there are a lot of example [here](https://github.com/argoproj/argo/tree/master/examples) too.
```
Comment (devonboyer) [inline test/e2e/README.md]: "...there are a lot of example**s**"

**Review by devonboyer — COMMENTED** [review]
This is great and will be super helpful!

File: test/e2e/README.md [hunk 3]
```diff
@@ -1,11 +1,124 @@
...
+      var nb = db.series.find({
+      metric: "kubernetes_state.deployment.replicas_available", 
+      tags: {$all: ["namespace:default", "deployment:fake-datadog"] }, 
+      "points.0.1": { $eq: 1} });      
+      print("find: " + nb)
+      if (nb != 0) {
+        break;
+      }
+      prevNb = nb;
+      sleep(30000);
```
Comment (devonboyer) [inline test/e2e/README.md]: This seems like a large value between retries: `sleep(30000)`.

A lot of our steps are using `sleep(2000)`.

What do you think?

Comment (JulienBalestra) [reply test/e2e/README.md]: As we discussed, a large value is needed to observe a removed check.
A short one is better on query, to speed the tests up.

**Review by devonboyer — APPROVED** [review]

**Pull request merged**